src/mscorlib/src/System/Array.cs

```diff
@@ -1073,6 +1073,42 @@ public static T[] Empty<T>()
             return Array.FindIndex(array, match) != -1;
         }
 
+        public static void Fill<T>(T[] array, T value)
+        {
+            if (array == null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
+            }
+
+            for (int i = 0; i < array.Length; i++)
+            {
+                array[i] = value;
+            }
```

Comment:
NOTE: This is just the initial implementation which gets things up and running. This can probably be optimized later to be specialized for different types, e.g. use `initblk` for bytes, or write 16 bytes at a time for other primitives. But that is work for a later PR.